timestamp-impl/src/main/java/com/palantir/timestamp/PersistentTimestampService.java

```diff
@@ -37,19 +38,20 @@ public PersistentTimestampService(AvailableTimestamps availableTimestamps, Execu
 
         this.availableTimestamps = availableTimestamps;
         this.executor = executor;
+        this.isAllocationTaskSubmitted = new AtomicBoolean(false);
     }
 
     public static PersistentTimestampService create(TimestampBoundStore tbs) {
         PersistentUpperLimit upperLimit = new PersistentUpperLimit(tbs);
         LastReturnedTimestamp lastReturned = new LastReturnedTimestamp(upperLimit.get());
         AvailableTimestamps availableTimestamps = new AvailableTimestamps(lastReturned, upperLimit);
-        ExecutorService executor = Tracers.wrap(
-                PTExecutors.newSingleThreadExecutor(
-                        PTExecutors.newThreadFactory("Timestamp allocator", Thread.NORM_PRIORITY, true)));
+        ExecutorService executor = PTExecutors.newSingleThreadExecutor(
+                PTExecutors.newThreadFactory("Timestamp allocator", Thread.NORM_PRIORITY, true));
```

Comment:
note to reviewers: removing tracing for now as it was allocating a bunch and is likely unnecessary.